cmd/aws-vpc-cni-init/main.go

```diff
@@ -41,45 +44,89 @@ const (
 	envEgressV6                 = "ENABLE_V6_EGRESS"
 )
 
-func getNodePrimaryIF() (string, error) {
-	var primaryIF string
-	primaryMAC, err := imds.GetMetaData("mac")
+func getNodePrimaryIFs() ([]string, error) {
+
+	var primaryIF []string
+
+	ctx := context.Background()
+
+	awsconfig, err := awssession.New(ctx)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to create aws session")
+	}
+	ec2Metadata := ec2metadata.NewFromConfig(awsconfig)
+	imds := awsutils.TypedIMDS{ec2Metadata}
+
+	macAddresses, err := imds.GetMACs(ctx)
```

Comment:
Won't this return all ENIs - i.e, even secondary ENIs - https://github.com/aws/amazon-vpc-cni-k8s/blob/0dccb2296cc25828b2772daae0f8297185b04686/pkg/awsutils/imds.go#L210

Reply:
Yes.. But we are filtering the ENIs which are at device-number 0..